core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -67,10 +68,15 @@ private void disableConnectivityReceiver() {
 
     private void setReceiverState(int receiverState) {
         if (USE_COMPONENT_ENABLED_SETTING) {
-            getPackageManager().setComponentEnabledSetting(new ComponentName(this, ConnectivityReceiver.class), receiverState, PackageManager.DONT_KILL_APP);
+            getPackageManager().setComponentEnabledSetting(connectivityReceiverComponent(), receiverState, PackageManager.DONT_KILL_APP);
         }
     }
 
+    @VisibleForTesting
+    protected ComponentName connectivityReceiverComponent() {
+        return new ComponentName(this, ConnectivityReceiver.class);
```

Comment:
you could use this constructor http://developer.android.com/reference/android/content/ComponentName.html#ComponentName(java.lang.String, java.lang.String) then you could make component name a static final field ... i don't think this should need to be overriden for test doesn't really make sense